framework/src/outputs/TableOutput.C

```diff
@@ -85,18 +85,22 @@ TableOutput::outputVectorPostprocessors()
   // Loop through the postprocessor names and extract the values from the VectorPostprocessorData storage
   for (const auto & vpp_name : out)
   {
-    const std::map<std::string, VectorPostprocessorValue*> & vectors = _problem_ptr->getVectorPostprocessorVectors(vpp_name);
+    const auto & vectors = _problem_ptr->getVectorPostprocessorVectors(vpp_name);
 
-    FormattedTable & table = _vector_postprocessor_tables[vpp_name];
+    auto table_it = _vector_postprocessor_tables.lower_bound(vpp_name);
```

Comment:
That's a fair criticism. I'm very, very tainted on RHS use of std::map::operator[]. I would say that I track down a bug about every 4-6 weeks that's related to careless use of that operator. I'd almost rather eradicate it's use anywhere in MOOSE but the syntax is very convenient which makes that hard to do. 

This particular case is actually one of the few cases where RHS brackets are OK because we want to always create tables so I'm torn. In general, this is the syntax I'd like to move to because for the normal access case, you either add a straight up error when the item isn't found or often just an assertion when you are "sure" the item is there (yeah right...)

Here's another crazy idea... We use maps a lot in MOOSE. Maybe it's time to think about replacing our use of maps with a better container. We could start with just aliasing our container to std::map but quickly move to hiding implementation details such as the code you see there behind familiar operators like brackets. I've been using this `lower_bound` code lately because it's the best way to check for an item and insert if if necessary because you can use `hint`. Using `find` doesn't work because you always end up with a iterator to the end of the map meaning you have to do two tree searches to handle the insert case. Is it premature optimization? Maybe, but maybe not. I'm starting to put these in as I fix accidental insertion.
</rant>